File: frontend/src/types/gasto.rs
use serde::{Deserialize, Serialize};

use crate::types::deserialize_f64_from_any;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
#[allow(clippy::struct_field_names)]
pub struct Gasto {
    pub id: String,
    pub propiedad_id: String,
    pub unidad_id: Option<String>,
    pub categoria: String,
    pub descripcion: String,
    #[serde(deserialize_with = "deserialize_f64_from_any")]
    pub monto: f64,
    pub moneda: String,
    pub fecha_gasto: String,
    pub estado: String,
    pub proveedor: Option<String>,
    pub numero_factura: Option<String>,
    pub notas: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct CreateGasto {
    pub propiedad_id: String,
    pub unidad_id: Option<String>,
    pub categoria: String,
    pub descripcion: String,
    pub monto: f64,
    pub moneda: String,
    pub fecha_gasto: String,
    pub proveedor: Option<String>,
    pub numero_factura: Option<String>,
    pub notas: Option<String>,
}

#[derive(Debug, Clone, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct UpdateGasto {
    pub categoria: Option<String>,
    pub descripcion: Option<String>,
    pub monto: Option<f64>,
    pub moneda: Option<String>,
    pub fecha_gasto: Option<String>,
    pub unidad_id: Option<String>,
    pub proveedor: Option<String>,
    pub numero_factura: Option<String>,
    pub estado: Option<String>,
    pub notas: Option<String>,
}

File: android/core/model/src/main/kotlin/com/propmanager/core/model/Gasto.kt
package com.propmanager.core.model

import java.math.BigDecimal
import java.time.Instant
import java.time.LocalDate

data class Gasto(
    val id: String,
    val propiedadId: String,
    val unidadId: String?,
    val categoria: String,
    val descripcion: String,
    val monto: BigDecimal,
    val moneda: String,
    val fechaGasto: LocalDate,
    val estado: String,
    val proveedor: String?,
    val numeroFactura: String?,
    val notas: String?,
    val createdAt: Instant,
    val updatedAt: Instant,
    val isPendingSync: Boolean = false
)
